dspace-api/src/main/java/org/dspace/content/RelationshipMetadataServiceImpl.java

```diff
@@ -0,0 +1,283 @@
+     * @param isLeftwards           Determines the direction of the virtual metadata
+     * @return                      The list of virtual metadata values
+     */
+    private List<RelationshipMetadataValue> findVirtualMetadataFromConfiguration(Context context, Item item,
+        HashMap<String, VirtualMetadataConfiguration> hashMaps, Item otherItem, String relationName,
+        Relationship relationship, int place, boolean isLeftwards) throws SQLException {
+
+        List<RelationshipMetadataValue> resultingMetadataValueList = new LinkedList<>();
+        for (Map.Entry<String, VirtualMetadataConfiguration> entry : hashMaps.entrySet()) {
+            String key = entry.getKey();
+            VirtualMetadataConfiguration virtualBean = entry.getValue();
+
+            if (virtualBean.getPopulateWithNameVariant()) {
+                String wardLabel = isLeftwards ? relationship.getLeftwardValue() : relationship.getRightwardValue();
+                if (wardLabel != null) {
+                    resultingMetadataValueList.add(
+                        constructRelationshipMetadataValue(context, item, relationship.getID(), place, key, virtualBean,
+                                                           wardLabel));
+                } else {
+                    resultingMetadataValueList.addAll(
+                            handleVirtualBeanValues(context, item, otherItem, relationship, place, key, virtualBean));
+                }
+            } else {
+                resultingMetadataValueList.addAll(
+                        handleVirtualBeanValues(context, item, otherItem, relationship, place, key, virtualBean));
+            }
+        }
+        return resultingMetadataValueList;
+    }
+
+    /**
+     * This method will retrieve a list of RelationshipMetadataValue objects based on the config passed along in the
+     * hashmaps parameter. The beans will be used to retrieve the values for the RelationshipMetadataValue objects
+     * and the keys of the hashmap will be used to construct the RelationshipMetadataValue object.
+     *
+     * @param context               The context
+     * @param item                  The item whose virtual metadata is requested
+     * @param otherItem             The related item whose actual metadata is requested
+     * @param relationship          The relationship whose virtual metadata is requested
+     * @param place                 The place to use in the virtual metadata
+     * @param key                   The key corresponding to the VirtualMetadataConfiguration
+     * @param virtualBean           The VirtualMetadataConfiguration object which will generate the
+     *                              virtual metadata. This configuration is applicable for a relationship
+     *                              between both items
+     * @return                      The list of virtual metadata values
+     */
+    private List<RelationshipMetadataValue> handleVirtualBeanValues(
```

Comment:
Another method name that could be improved, as it doesn't seem to describe the actions of the method.  Maybe `findRelationshipMetadataValueFromBean()`?

Reply:
I've renamed the method, the name you suggested is indeed good